src/js/components/AppPageComponent.jsx

```diff
@@ -273,15 +282,148 @@ var AppPageComponent = React.createClass({
     );
   },
 
+  getVolumeStatus: function () {
+    var volume = this.getVolume();
+
+    if (volume == null) {
+      return null;
+    }
+
+    var className = classNames("volume-status", {
+      "volume-attached": volume.status === "Attached"
+    });
+
+    return (
+      <h2 className={className}>{volume.status}</h2>
+    );
+  },
+
+  getVolume: function () {
+    var {app, volumeId} = this.state;
+
+    if (app.tasks != null) {
+      return null;
+    }
+
+    var tasks = app.tasks;
+    return tasks
+      // Get the first volume from a task with the same id as provided
+      // by the router. This should be unique.
+      .reduce((memo, task) => {
+        if (memo == null) {
+          return task.localVolumes
+            .filter(volume => volume.persistenceId === volumeId)
+            .reduce((memo, volume) => {
+              if (memo != null) {
+                return memo;
+
+              }
+              volume.taskId = task.id;
+              volume.host = task.host;
+              volume.status = task.status == null
+                ? "Detached"
+                : "Attached";
+              return volume;
+            }, null);
+        }
+        return memo;
+      }, null);
+  },
+
+  getVolumeDetails: function () {
```

Comment:
I think it makes sense to move this into an own compontent.
See getTaskDetailComponent() as an example in this file.